CMake: Remove undefined function

rwalton-arm opened the pull request:
The function mbed_set_mbed_target_linker_script no longer exists. This
commit removes the call to this function from the application
CMakeLists.txt.

Fixes https://github.com/ARMmbed/mbed-os-example-socket-stats/pull/73

rwalton-arm committed:
CMake: Remove undefined function

The function mbed_set_mbed_target_linker_script no longer exists. This
commit removes the call to this function from the application
CMakeLists.txt.

0xc0170 commented:
I updated the destination branch to development (master is the release branch)

0xc0170 commented:
@rwalton-arm Can you also add a commit to fix Travis so we get all green? https://github.com/ARMmbed/mbed-os-example-socket-stats/blob/master/.travis.yml#L82 - this needs an update

```
        - mbedtools deploy
        - echo mbedtools compile -t GCC_ARM -m ${TARGET_NAME} -b ${PROFILE}
        - mbedtools compile -t GCC_ARM -m ${TARGET_NAME} -b ${PROFILE}
```

rwalton-arm committed:
travis: Update mbedtools invocation

Travis was using `mbed-tools build` instead of `mbed-tools compile`.

Update travis.yml to use the correct command.

0xc0170 commented:
Looks better but travis still fails, can you review? It complains about the linker. I've checked config path, looks good. What else we have missed?

rwalton-arm commented:
> Looks better but travis still fails, can you review? It complains about the linker. I've checked config path, looks good. What else we have missed?

Yeah I just saw the failure. Working locally. Let me check what the problem is...

0xc0170 commented:
mbed-os.lib on devel branch can be master (no sha), is this the case we are fetching old version?

0xc0170 commented:
https://github.com/ARMmbed/mbed-os-example-socket-stats/blob/development/mbed-os.lib - I assume this is it, pointing to old release.

rwalton-arm commented:
> https://github.com/ARMmbed/mbed-os-example-socket-stats/blob/development/mbed-os.lib - I assume this is it.

Yep, that's it. I have a later copy of mbed-os locally.

rwalton-arm commented:
> https://github.com/ARMmbed/mbed-os-example-socket-stats/blob/development/mbed-os.lib - I assume this is it, pointing to old release.

Looks like [this PR](https://github.com/ARMmbed/mbed-os-example-socket-stats/pull/65) fixes the lib file. Should we merge it?

0xc0170 commented:
Merged, lets rebas this one now

rwalton-arm commented:
> Merged, lets rebas this one now

Rebased. CI looks green now.

LDong-Arm reviewed (changes requested):
Thanks for the change. Please also add instructions for Mbed CLI 2 to README, https://github.com/ARMmbed/mbed-os-example-blinky can be used as a template.

rwalton-arm committed:
readme: Add instructions for "Mbed CLI 2"

Update the README.md with instructions on how to build with mbed-tools.

rwalton-arm commented:
> Thanks for the change. Please also add instructions for Mbed CLI 2 to README, https://github.com/ARMmbed/mbed-os-example-blinky can be used as a template.

I've updated README.md in 949b404. Please review.

LDong-Arm commented on README.md:
```diff
@@ -7,7 +7,16 @@ Please refer to the [mbed-os-example-sockets](https://github.com/ARMmbed/mbed-os
 You can build the project with all supported [Mbed OS build tools](https://os.mbed.com/docs/mbed-os/latest/tools/index.html). However, this example project specifically refers to the command-line interface tool [Arm Mbed CLI](https://github.com/ARMmbed/mbed-cli#installing-mbed-cli).
 
-1. Install Mbed CLI.
+## Mbed OS build tools
```
Line 8 above needs to be either updated or removed, as it links to the old tools only.

rwalton-arm replied:
This is also the case in https://github.com/ARMmbed/mbed-os-example-blinky/blob/development/README.md - we will also need to fix it there.

LDong-Arm replied:
Yep, there are just many examples to maintain. Before we find a way to combine examples or factorise docs, I guess keeping things short & simple is the best.

LDong-Arm reviewed (commented):
Thanks, just a small comment

LDong-Arm reviewed (approved):
Thanks, LGTM
